V2.0 changes, breaks API backwards compatibility.

matthewpersico opened the pull request.

matthewpersico committed:
All functions now contain "kvstore" as part of the name in order to not pollute global namespace

matthewpersico committed:
API change, version 2.0

- 'ls' now only lists namespaces
- 'keys' lists keys of namespace
- 'vals' lists vals of namespace
- -v, --version added to print version

matthewpersico committed:
Fixed 'ls' command to ignore .lock files.

matthewpersico committed:
Updated for V2.0 API changes, updated license info.

ccarpita commented on LICENSE.md:
```diff
@@ -0,0 +1,19 @@
+Copyright 2017 Chris Carpita
```
I've labelled as MIT license recently (see README) to clarify rights for contributors, so this file should be consistent.  Thanks for adding though!

matthewpersico commented:
I added the file explicitly because there are systems that look for it in
order to automatically figure out if it will be allowed in house at
commercial shops. Like mine. :-)

ccarpita commented on kvstore.sh:
```diff
@@ -331,3 +347,9 @@ kvstore () {
+## Local Variables:
+## sh-basic-offset: 2
+## sh-indentation: 2
+## End:
```
Is it possible for emacs to read these settings on a project/directory level?   Would prefer to keep editor-specific code out, but then again you're the biggest contributor, so we can keep it if it needs to be file-level.

matthewpersico replied:
Yes. I did some research and found that I could add a .dir-locals.el file to the directory and stuff those items in there. I'll amend the commit this evening,

ccarpita commented:
@matthewpersico understood about the license file, which is a good idea.  I realize now that it is the MIT license in fact.  No need to change anything 👍

ccarpita commented:
That solution sounds good to me

On Thu, Aug 10, 2017 at 8:17 AM Matthew O. Persico <notifications@github.com>
wrote:

> *@matthewpersico* commented on this pull request.
> ------------------------------
>
> In kvstore.sh
> <https://github.com/ccarpita/kvstore/pull/5#discussion_r132437152>:
>
> > @@ -331,3 +347,9 @@ kvstore () {
>  }
>  kvstore "$@"
>  exit $?
> +
> +## for emacs...
> +## Local Variables:
> +## sh-basic-offset: 2
> +## sh-indentation: 2
> +## End:
>
> Yes. I did some research and found that I could add a .dir-locals.el file
> to the directory and stuff those items in there. I'll amend the commit if
> that solution is ok with you.
>
> —
> You are receiving this because you commented.
>
>
> Reply to this email directly, view it on GitHub
> <https://github.com/ccarpita/kvstore/pull/5#discussion_r132437152>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/ABjrzQXz_qFPDT52Hl14oCGVW0yg8aacks5sWvTvgaJpZM4Oy1F1>
> .
>

matthewpersico committed:
Moved emacs config out of main file

matthewpersico commented:
Done. Emacs config moved out to .dir-locals.el.

matthewpersico committed:
Prevent namespaces from ending in '.lock'.

Prevents conflicts with .lock files produced by flock. Plus, some minor
format nits.

ccarpita commented:
👍
